browser/ui/brave_ads/ad_notification_popup.cc

```diff
@@ -0,0 +1,309 @@
+/* Copyright (c) 2020 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include "brave/browser/ui/brave_ads/ad_notification_popup.h"
+
+#include <map>
+#include <memory>
+#include <utility>
+
+#include "base/bind.h"
+#include "brave/browser/profiles/profile_util.h"
+#include "brave/browser/ui/brave_ads/ad_notification_view.h"
+#include "brave/browser/ui/brave_ads/ad_notification_view_factory.h"
+#include "brave/browser/ui/brave_ads/bounds_util.h"
+#include "brave/components/brave_ads/common/pref_names.h"
+#include "build/build_config.h"
+#include "components/prefs/pref_service.h"
+#include "ui/display/screen.h"
+#include "ui/gfx/geometry/point.h"
+#include "ui/gfx/geometry/rect.h"
+#include "ui/gfx/geometry/size.h"
+#include "ui/views/layout/fill_layout.h"
+#include "ui/views/metadata/metadata_impl_macros.h"
+#include "ui/views/widget/widget.h"
+
+#if defined(OS_WIN)
+#include "ui/views/widget/desktop_aura/desktop_native_widget_aura.h"
+#endif
+
+namespace brave_ads {
+
+namespace {
+
+// TODO(https://github.com/brave/brave-browser/issues/14957): Decouple
+// AdNotificationPopup management to NotificationPopupCollection
+std::map<std::string, AdNotificationPopup*> g_ad_notification_popups;
+
+const int kPopupPadding = 10;
+
+}  // namespace
+
+AdNotificationPopup::AdNotificationPopup(Profile* profile,
+                                         const AdNotification& ad_notification)
+    : profile_(profile),
+      ad_notification_(ad_notification),
+      ad_notification_view_(
+          AdNotificationViewFactory::Create(ad_notification_)) {
```

Comment:
this does not appear to get destroyed anywhere

Reply:
`ad_notification_view_` child view is added to and owned by the view.  See `AdNotificationPopup::CreatePopup()`